wasm2c/wasm-rt-impl.c

```diff
@@ -351,6 +405,7 @@ bool wasm_rt_is_initialized(void) {
 }
 
 void wasm_rt_free(void) {
+  WASM_RT_MUTEX_LOCK(g_func_type_mutex);
```

Comment:
Presumably is reasonable to assert that there are no active instances when this is called  (since all best would be off if they were)?  So technically maybe the lock isn't needed here?

Reply:
Yup, I considered not having it as well. I don't have a strong preference one way or the other. Let me know if you prefer not having it.

Reply:
No string preference here, just confirming my assumptions.  Maybe add a comment?